app/src/main/java/com/kms/cura/view/fragment/RegDoc_ProfessionalFragment.java

```diff
@@ -0,0 +1,213 @@
+        speciality.add("A");
+        speciality.add("B");
+        speciality.add("C");
+
+        facility = new ArrayList<>();
+        facility.add("A");
+        facility.add("B");
+        facility.add("C");
+        facility.add("D");
+        facility.add("E");
+        facility.add("F");
+        facility.add("G");
+        facility.add("H");
+        // ---------------------------------------
+        updateSpinner = this;
+        degree.add(spnText[0] + "*");
+        speciality.add(spnText[1] + "*");
+        facility.add(spnText[2] + "*");
+        btnBack = setupButton(myFragmentView, R.id.btnRegisterBack, this);
+        btnNext = setupButton(myFragmentView, R.id.btnRegisterNext, this);
+
+        reformData();
+
+        adapterDegree = new StringListAdapter(getActivity(), R.layout.string_list_item, degree, checkedDegree);
+        adapterSpeciality = createAdapter(speciality, checkedSpeciality);
+        adapterFacility = createAdapter(facility, checkedFacility);
+        spnDegree = (Spinner) myFragmentView.findViewById(R.id.spnDegree);
+        spnDegree.setAdapter(adapterDegree);
+        spnDegree.setSelection(checkedDegree);
+        spnSpeciality = setupSpinner(myFragmentView, R.id.spnSpeciality, adapterSpeciality);
+        spnFacility = setupSpinner(myFragmentView, R.id.spnFacility, adapterFacility);
+        return myFragmentView;
+    }
+
+    private Spinner setupSpinner(View parent, int resourceId, CheckBoxAdapter adapter) {
+        Spinner spn = (Spinner) parent.findViewById(resourceId);
+        spn.setAdapter(adapter);
+        spn.setSelection(adapter.getCount());
+        return spn;
+    }
+
+    private CheckBoxAdapter createAdapter(ArrayList<String> data, boolean[] checked) {
+        return new CheckBoxAdapter(getActivity(), R.layout.check_box_item, data, checked, updateSpinner);
+    }
+
+    private ImageButton setupButton(View parent, int resourceId, View.OnClickListener listener) {
+        ImageButton btn = (ImageButton) parent.findViewById(resourceId);
+        btn.setOnClickListener(listener);
+        return btn;
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.btnRegisterNext) {
+            if (checkInput()) {
+                // to next page
+                Bundle bundle = createBundle();
+                Fragment account = new RegDoc_AccountFragment();
+                account.setArguments(bundle);
+                getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, account).commit();
+            }
+        } else if (v.getId() == R.id.btnRegisterBack) {
+            Bundle bundle = createBundle();
+            Fragment personal = new RegDoc_PersonalFragment();
+            personal.setArguments(bundle);
+            getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, personal).commit();
+        }
+    }
+
+    private boolean checkInput() {
```

Comment:
this method is too long and have repetitive code. Should re-factor and re-use code here